### PR TITLE
LPS-71381 As a developer, I would like to be able to get System-scoped configurations using the ConfigurationProviderUtil

### PR DESCRIPTION
**Pull request by drewbrokke**
This is an update for [LPS-71381](https://issues.liferay.com/browse/LPS-71381).

Copied from Slack:

> Hey @jorgeferrer, I just made a ticket to add getting a System scoped Configuration using the ConfigurationProviderUtil.java. I think it would be very beneficial for us to have this method of getting System scoped Configs available to us while we finish our properties migrations, particularly in classes that are in modules, but are not components.  Changing the constructor is not always an option for us, so having this alternative would speed things up.

/cc @pei-jung

### CODE REVIEW

File: portal-kernel/src/com/liferay/portal/kernel/settings/GlobalServiceSettingsLocator.java [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+	@Override
+	public Settings getSettings() throws SettingsException {
+		Settings configurationBeanSettings =
+			_settingsLocatorHelper.getConfigurationBeanSettings(
+				_configurationPid);
+
+		return _settingsLocatorHelper.getGlobalPortalPreferenceSettings(
+			configurationBeanSettings);
+	}
```
Comment (pei-jung) [inline portal-kernel/src/com/liferay/portal/kernel/settings/GlobalServiceSettingsLocator.java]: can we just return the configurationBeanSettings here? I'm not sure how settings work, but do we really need to get the settings from preferences if it's a system configuration?

**Comment by JorgeFerrer** [comment]
Just started reviewing :) <br><br>:octocat: *Sent from [GH](http://nodegh.io).*

File: portal-kernel/src/com/liferay/portal/kernel/settings/GlobalServiceSettingsLocator.java (hunk 1 above)
Comment (JorgeFerrer) [inline portal-kernel/src/com/liferay/portal/kernel/settings/GlobalServiceSettingsLocator.java]: Yeah, it should be enough. We should not use the preferences table for System wide configuration.

File: portal-kernel/src/com/liferay/portal/kernel/settings/GlobalServiceSettingsLocator.java [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.portal.kernel.settings;
+
+/**
+ * @author Drew Brokke
+ */
+public class GlobalServiceSettingsLocator implements SettingsLocator {
```
Comment (JorgeFerrer) [inline portal-kernel/src/com/liferay/portal/kernel/settings/GlobalServiceSettingsLocator.java]: Why not call it SystemServiceSettingsLocator?

File: portal-kernel/src/com/liferay/portal/kernel/settings/SettingsLocatorHelper.java [hunk 3]
```diff
@@ -35,6 +35,8 @@ public Settings getCompanyPortletPreferencesSettings(
 	public Settings getConfigurationBeanSettings(
 		String configurationPid, Settings parentSettings);
 
+	public Settings getGlobalPortalPreferenceSettings(Settings parentSettings);
```
Comment (JorgeFerrer) [inline portal-kernel/src/com/liferay/portal/kernel/settings/SettingsLocatorHelper.java]: I think this should be getSystemPreferenceSettings() for consistency with the terminology used across our documentation

**Comment by JorgeFerrer** [comment]
Hey Drew, I've made a few suggestions. 

Other than that it looks good. Thanks for the proactivity!

**Comment by drewbrokke** [comment]
Thanks for the review, @JorgeFerrer.

One other question - is it worth it to also validate that the requested class is indeed a System-scoped configuration?  What if a dev calls the method with a class from a different scope?

**Comment by JorgeFerrer** [comment]
Good question. I have mixed thoughts about it. On one side I think it's a good idea, but it would need to be applied for all scopes, not just system (and that could break some things).

On the other hand, what if you want to obtain the system scoped configuration to read the defaults even if the configuration is scoped per company?

**Comment by drewbrokke** [comment]
That's a good point.  I suppose we could leave this up to the developer to use each method as they feel it's necessary.  Since the others are not scope-checked already, I think we could leave it for now since the risks are pretty minimal.

**Pull request closed without merging**